### ISSUE TITLE
Machine Translation Task with DiffuSeq

### ISSUE THREAD

**Issue by chiral-carbon** [Shark-NLP/DiffuSeq#74]
Hi @summmeer, 

I was wondering how I might go about implementing a machine translation task with DiffuSeq. I have trained DiffuSeq for the paraphrase task, but I want to be able to use it for translation tasks. 
Would supplying a translation dataset to the existing codebase (since it designed for seq2seq tasks) suffice or would further changes be required? 

Would appreciate any advice, thanks!

**Comment by summmeer** [comment]
Hi,
You can have a try. But different hyper-parameters may lead to different results, including bsz, steps, dim, seq_len, and tokenizers. Currently many follow-up works achieve better MT performance and you can refer to their codebase, too.

**Comment by chiral-carbon** [comment]
Yeah makes sense, thanks! Are you referring to works like SeqDiffuSeq which builds on DiffuSeq directly?

**Comment by summmeer** [comment]
It depends on what your goal is using diffusion model for MT tasks. Follow-up works are not exactly the same with DiffuSeq. SeqDiffuSeq is based on encoder-decoder architecture, while [RDM](https://github.com/HKUNLP/reparam-discrete-diffusion) is based on discrete text diffusion. [This work](https://arxiv.org/pdf/2308.12219v2.pdf) also involves pre-trained MLMs. If you're aiming the performance, you could refer to the SOTA model.

**Comment by chiral-carbon** [comment]
@summmeer thanks, this is very helpful! in the paper [DiNoiSer](https://arxiv.org/pdf/2302.10025.pdf), the authors claim to have surpassed DiffuSeq's performance on the WMT14 EN->DE task, so I wanted to do a similar comparison between DiffuSeq and DiNoiSer on the IWSLT14 task, but DiffuSeq takes a long time to train. 
Even with the QQP task reported in the paper, I tried training it to replicate the results and on 4 A100 GPUs it took 6.5 days to train ([WandB overview](https://wandb.ai/dlpros/diffuseq/runs/l75snzgl/overview?workspace=user-)), so do you think there is additional distributed training code required to train DiffuSeq more efficiently? 

Sorry for the trivial question, your replies are really helpful, thanks!

**Comment by summmeer** [comment]
Hi,
Maybe you can try our updated version 2, which is 4x faster on training and 800x faster on sampling on QQP datasets. [We update the information of v2 in README.md]

**Comment by chiral-carbon** [comment]
I will, thanks a lot!
